### ISSUE TITLE
roof and offset issue for some combinations

### ISSUE THREAD

**Issue by UBaer21** [openscad/openscad#4120]
openSCAD version 2022.02.04

skeleton error when preview:
"ERROR: Skeleton computation error. /root/workspace/src/roof_vd.cc line 374: Voronoi error in file , line 9"

`for(i=[2:4],fn=[75:77])translate([i*10,(fn-75)*10])roof()offset(i/10,$fn=fn)difference(){square(5);circle(4,$fn=6);}`
When using  offset (.3,$fn=76)  - other shapes will show problems for other combinations of offset and $fn
![image](https://user-images.githubusercontent.com/93376232/153823516-31c9fcbb-32db-4424-8137-0244097baefb.png)

CGAL Errors when render
"ERROR: CGAL error in CGAL_Nef_polyhedron3(): CGAL ERROR: assertion violation!
Expr: e_below != SHalfedge_handle()
File: /mxe/usr/x86_64-w64-mingw32.static.posix/include/CGAL/Nef_3/SNC_FM_decorator.h
Line: 418"

```
for(i=[0:10],fn=[93:120])union(){
translate([i*10,(fn-100)*10])roof()offset(i/10,$fn=fn)difference(){square(5);square(4);}
cube(.001);
}
```

![image](https://user-images.githubusercontent.com/93376232/153825007-4e2baad2-d62b-4bb1-a704-a7a338088cc8.png)

**Comment by khumarahn** [comment]
https://github.com/openscad/openscad/pull/4240

**Comment by khumarahn** [comment]
To the best of my understanding, this issue has two parts. I fixed the one related to
```ERROR: Skeleton computation error. /root/workspace/src/roof_vd.cc line 374: Voronoi error in file , line 9```

But
```Expr: e_below != SHalfedge_handle()```
is the F5-before-F6 problem: it can be worked around by cleaning the caches and pressing F6 right away before F5. In the particular example above, the offset shape has with many segments which are close to a circle
![screenshot-20220501@114046](https://user-images.githubusercontent.com/5206045/166140553-cd6b6eeb-857c-4f0e-bd55-3d946d3b6771.png)
This leads to many close-by vertices in the Voronoi diagram. The renderer (F5) snaps these to a grid, and this destroys the geometry. I'm not sure how to solve this.

**Comment by t-paul** [comment]
Yes, the F5 before F6 issue is separately tracked via #2991 and may need some cache rework.
